File: scrapper/formatter.js
'use strict';

const fs = require('fs');

let rawdata = fs.readFileSync('products_sample.json');
let products = JSON.parse(rawdata);
var prod_fin = []
for (var i = 0; i < products.length; i++) {
    
    if(products[i].desc){
    var desc_text = products[i].desc
            desc_text = desc_text.replace(/<style.*?\/ *?style>/gms , '')
            desc_text = desc_text.replace(/< *?script(?:.*?)< *?\/ *?script>/gsm,'')
            desc_text = desc_text.replace(/<br>|<br\/>|<br \/>/gm,'\r\n')
            desc_text = desc_text.replace(/<.*?>/gm,' ')
            desc_text = desc_text.replace(/\\t/gm,'')
            desc_text = desc_text.replace(/\t/gm,'')
            desc_text = desc_text.replace(/(?: *\\n){2,}/gm,' \r\n')
            desc_text = desc_text.replace(/(?: *\n){2,}/gm,' \r\n')
            desc_text = desc_text.replace(/(?: *\r\n){2,}/gm,' \r\n')
            desc_text = desc_text.replace(/(?: *\\r\\n){2,}/gm,' \r\n')
            desc_text = desc_text.replace(/ {2,}/gm,' ')
            products[i].desc_text = desc_text
            prod_fin.push(products[i])
}
}

let data = JSON.stringify(prod_fin);
fs.writeFileSync('products_reparsed.json', data);

File: scrapper/formatter.py
import re
import json



with open('products_sample.json') as json_file:
    data = json.load(json_file)
    f = open("products_reparsed.json", "a")
    f.write('[')
    # Print the type of data variable
    for i in range(0,len(data)):
        desc_text = data[i]['desc']
        if desc_text:
            
            #desc_text = re.sub(r'<style type=\"text/css\">.*?/style>','', desc_text)
            #desc_text = re.sub(r'<style>.*?</style>','', desc_text)
            #desc_text = re.sub(r'<style.*?/style>','', desc_text)
            desc_text = re.sub(re.compile('<style.*?/ *?style>', re.DOTALL),'', desc_text)
            desc_text = re.sub(re.compile('< *?script(?:.*?)< *?/ *?script>', re.DOTALL),'', desc_text)
            desc_text = re.sub(r'<br>|<br\/>|<br \/>','\r\n', desc_text)
            desc_text = re.sub(r'<.*?>',' ', desc_text)
            desc_text = re.sub(r'\\t','', desc_text)
            desc_text = re.sub(r'\t','', desc_text)
            desc_text = re.sub(r'(?: *\\n){2,}',' \r\n', desc_text)
            desc_text = re.sub(r'(?: *\n){2,}',' \r\n', desc_text)            
            desc_text = re.sub(r'(?: *\r\n){2,}',' \r\n', desc_text)
            desc_text = re.sub(r'(?: *\\r\\n){2,}',' \r\n', desc_text)
            desc_text = re.sub(r' {2,}',' ', desc_text)
            data[i]['desc_text'] = desc_text
            if desc_text and desc_text != ' ':
                
                json.dump(data[i], f)
                f.write(',\n')
                
    f.write(']')
    f.close()